cmd/kubeops/internal/handler/handler.go

```diff
@@ -72,19 +72,31 @@ func WithHandlerConfig(storageForDriver agent.StorageForDriver, options Options)
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
-			kubeClient, err := kubernetes.NewForConfig(restConfig)
+			userKubeClient, err := kubernetes.NewForConfig(restConfig)
 			if err != nil {
 				// TODO log details rather than return potentially sensitive details in error.
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
-			appRepoClient, err := appRepo.NewForConfig(restConfig)
+			svcRestConfig, err := rest.InClusterConfig()
 			if err != nil {
 				// TODO log details rather than return potentially sensitive details in error.
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
-			actionConfig, err := agent.NewActionConfig(storageForDriver, restConfig, kubeClient, namespace)
+			svcKubeClient, err := kubernetes.NewForConfig(svcRestConfig)
```

Comment:
Let's document right above this line why we're creating a separate client using service credentials for the chart client.